lib/rucio/common/hash_algorithms/default.py

```diff
@@ -0,0 +1,110 @@
+import hashlib
+import io
+import mmap
+import zlib
+
+from functools import partial
+
+from . import RucioHashAlgorithm
+
+
+class Adler32(RucioHashAlgorithm):
+    @staticmethod
+    def compute_on_file(file):
+        """
+        An Adler-32 checksum is obtained by calculating two 16-bit checksums A and B
+        and concatenating their bits into a 32-bit integer. A is the sum of all bytes in the
+        stream plus one, and B is the sum of the individual values of A from each step.
+
+        :param file: file name
+        :returns: Hexified string, padded to 8 values.
+        """
+
+        # adler starting value is _not_ 0
+        adler = 1
+
+        can_mmap = False
+        try:
+            with open(file, "r+b") as f:
+                can_mmap = True
+        except:
+            pass
+
+        try:
+            # use mmap if possible
+            if can_mmap:
+                with open(file, "r+b") as f:
+                    m = mmap.mmap(f.fileno(), 0)
+                    # partial block reads at slightly increased buffer sizes
+                    for block in iter(partial(m.read, io.DEFAULT_BUFFER_SIZE * 8), b""):
+                        adler = zlib.adler32(block, adler)
+            else:
+                with open(file, "rb") as f:
+                    # partial block reads at slightly increased buffer sizes
+                    for block in iter(partial(f.read, io.DEFAULT_BUFFER_SIZE * 8), b""):
+                        adler = zlib.adler32(block, adler)
+
+        except Exception as e:
+            raise Exception(
```

Comment:
nit: Should we define our own Exception? e.g. `HashAlgorithmException`

Reply:
I'm not against the idea - but note that this particular code snippet has been moved verbatim from rucio/common/utils.py.